File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F14.P1.java
class Main {
    // BEGIN PROBE F14.P1
    sealed interface Shape {}

    record Circle(double r) implements Shape {}

    record Rect(double w, double h) implements Shape {}

    static Shape newShape() {
        Shape s = new Circle(2.0);
        return s;
    }
    // END PROBE F14.P1

    public static void main(String[] args) {
        System.out.println(newShape());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F14.P1.kt
// BEGIN PROBE F14.P1
sealed interface Shape

class Circle(val r: Double) : Shape

class Rect(val w: Double, val h: Double) : Shape

val s: Shape = Circle(2.0)
// END PROBE F14.P1

fun main() {
    println(s is Circle)
}
